### DISCUSSION TITLE
Is there a hover option?

### DISCUSSION THREAD

**Discussion by yu0307** [ueberdosis/tiptap#615] [General Questions]
Is there a way for us to listen on hover(mouse over) event on the first level child elements under the root container? 
i was trying to have the menu visible when the element is hovering and hidden when mouse out. 

editor-content > p > other tages. 
listen for mouseover event on P only. 

Thank you

**Comment by peterkruty** [comment]
I'd like to know this too, any idea what is ideal way  to solve it?

**Comment by lorenjohnson** [comment]
+1

**Comment by thefinnomenon** [comment]
+1. This would be a really nice feature to have built-in to the editor.

**Comment by Priestch** [comment]
You can implement the hover feature use [ProseMirror Plugins](https://tiptap.dev/guide/custom-extensions#access-the-prosemirror-api).
View more from the official prosemirror [Plugin](https://prosemirror.net/docs/ref/#state.Plugin_System) doc.
Supported [events](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.htmlelementeventmap.html).

```javascript
import { Extension } from '@tiptap/core'
import { Plugin, PluginKey } from 'prosemirror-state'

export const HoverExtension = Extension.create({
  name: 'hover',

  addProseMirrorPlugins() {
    return [
      new Plugin({
        key: new PluginKey('hover'),
        props: {
          handleDOMEvents: {
            mouseover(view, event) {
              // do whatever you want
            }
          }
        },
      }),
    ]
  },
})
```

**Reply by rajatkulkarni95** [reply]
Quick question: What would you return inside the `mouseover` function since it only returns a bool or a void. The idea is to show a tooltip with anchor's href when hovering over an anchor element.

**Reply by emilianoyaryura** [reply]
any updates on this? 🙏🏼

**Reply by Priestch** [reply]
You can implement the tooltip in the mouseover event handler. The return value only show whether you already handle the mouseover event properly.

**Reply by emilianoyaryura** [reply]
but how can I get the anchors href of the hovered elements? such as [notion](https://notion.so/) does

**Reply by Priestch** [reply]
You can use [posAtDOM](https://prosemirror.net/docs/ref/#view.EditorView.posAtDOM) to get current pos from event.target, then use [nodeAt](https://prosemirror.net/docs/ref/#model.Node.nodeAt) based on pos to get the current node, when the node is your anchor node, you can get the href attribute from the node.

**Reply by emilianoyaryura** [reply]
Sorry for bothering you. I'm not quite understanding the flow... I'm getting this error: `RangeError: Position -1 outside of fragment (<paragraph("hola como andas? todo bien? aca te dejo el linaaaak de mi portfolio")>)`
```
 const handleLinkHover = React.useCallback(
    (event: React.MouseEvent<HTMLDivElement>) => {
      if (!editor) return
      const target = event.target as HTMLElement
      const view = editor.view as EditorView
      const pos = view.posAtDOM(target, 0)

      if (pos === null) return

      const node = view.state.doc.nodeAt(pos)
      if (!node || !node.isAtom) return

      if (node.type.name === 'link') {
        const href = node.attrs.href

        setLinkHover(href)
      }
    },
    [editor]
  )
```

```
    <div onMouseEnter={handleLinkHover}>
      {editor && menuBar && <MenuBar editor={editor} />}
      <EditorContent
        {...rest}
        className={clsx(
          '',
          className
        )}
        editor={editor}
      />
    </div>
```

**Reply by Priestch** [reply]
The error shows that the value of pos is invalid, does the node <paragraph("hola como andas? todo bien? aca te dejo el linaaaak de mi portfolio")> belongs to the rich editor content?

**Reply by emilianoyaryura** [reply]
@Priestch thanks for the help, I found the error. I was supposed to listen to the `onMouseOver` event instead of the `onMouseEnter`

**Reply by emilianoyaryura** [reply]
@Priestch sorry for bothering you again haha. Have you ever implemented tooltips? Because I don't find any documentation or example of how to, after getting the current `HTMLElement` (in my code [example](https://github.com/ueberdosis/tiptap/discussions/615#discussioncomment-6287120)) targeting it to render a tooltip below that link.

Thanks!

**Reply by Priestch** [reply]
@emilianoyaryura The prosemirror has an official [tooltip example](https://prosemirror.net/examples/tooltip/), you can also refer to the [lint example](https://prosemirror.net/examples/lint/) to see something similar to tooltip.
